### PR TITLE
Support checking for multiple consecutive lines in the expected test suite

### PR DESCRIPTION
**Pull request by zhassan-aws**
### Description of changes: 

**This is an alternative to #890**

Add support for checking for multiple consecutive lines appearing in the output for the expected/cargo-kani test suites. A backslash at the end of a line in the expected file is a marker that indicates that the next line should appear on a **consecutive** line in the output, e.g. for the following `expected` file:
```
SUCCESS\
Description: "index out of bounds: the length is less than or equal to the given index"
```
the test will only pass if there's a line that contains `SUCCESS` that is immediately followed by a line that contains `Description: "index out of bounds: the length is less than or equal to the given index"`.

This is a workaround until we do a cleaner rework of our verification result testing (https://github.com/model-checking/kani/issues/855).

### Resolved issues:

Resolves #ISSUE-NUMBER

### Testing:

* How is this change tested? Enhanced checking done in some of the expected tests

* Is this a refactor change?

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -381,18 +381,22 @@ impl<'test> TestCx<'test> {
         let output = proc_res.stdout.to_string() + &proc_res.stderr;
         if let Some(line) = TestCx::contains_lines(&output, expected.split('\n').collect()) {
             self.fatal_proc_rec(
-                &format!("test failed: expected output to contain the line: {}", line),
+                &format!("test failed: expected output to contain the line: {:?}", line),
```
Comment (celinval) [inline tools/compiletest/src/runtest.rs:384]: Is this needed?

File: tools/compiletest/src/runtest.rs [hunk 2]
```diff
@@ -381,18 +381,22 @@ impl<'test> TestCx<'test> {
...
     /// Otherwise, it returns the first line not found in `str`.
-    fn contains_lines<'a>(str: &str, lines: Vec<&'a str>) -> Option<&'a str> {
+    fn contains_lines<'a>(str: &str, lines: Vec<&'a str>) -> Option<String> {
         for line in lines {
-            if !str.contains(line) {
-                return Some(line);
+            // A "\\" pattern in the expected file is a marker that is used for
+            // separating multiple consecutive lines in the output, so
+            // substitute those markers with newlines
+            let consecutive_lines = line.replace("\\\\", "\n");
```
Comment (celinval) [inline tools/compiletest/src/runtest.rs:397]: Can you please still add a commy that this is a temporary fix?

**Review by celinval — APPROVED** [review]
Please address the comments before pushing. Thanks!

**Comment by celinval** [comment]
Nit: I would also prefer to use \ at the end of the line to specify that the pattern continues in the next line. But not a blocker since this is a temporary solution

**Comment by zhassan-aws** [comment]
@celinval I made two changes:
1. Switched to a (single) backslash at the end of a line to indicate that the next line should appear on a consecutive line in the output
2. Changed the check to be substring-based, i.e. the consecutive lines do not have to match exactly, but can each be a substring of a set of consecutive lines.

**Review by celinval — APPROVED** [review]
Amazing! Thanks

**Pull request merged**